streams/shared/src/main/scala/zio/stream/ZChannel.scala

```diff
@@ -1942,20 +1942,20 @@ object ZChannel {
     unwrapScopedWith { scope =>
       for {
         input         <- SingleProducerAsyncInput.make[InErr, InElem, InDone]
+        fiberId       <- ZIO.fiberId
         incoming       = ZChannel.fromInput(input)
-        n             <- ZIO.succeed(n)
-        bufferSize    <- ZIO.succeed(bufferSize)
-        mergeStrategy <- ZIO.succeed(mergeStrategy)
-        outgoing      <- Queue.bounded[ZIO[Env, Either[OutErr, OutDone], OutElem]](bufferSize)
+        n0             = n.toLong
+        bufferSize0    = bufferSize
+        mergeStrategy0 = mergeStrategy
+        outgoing       = Queue.unsafe.bounded[ZIO[Env, Either[OutErr, OutDone], OutElem]](bufferSize0, fiberId)(Unsafe.unsafe)
+        cancelers      = Queue.unsafe.unbounded[Promise[Nothing, Unit]](fiberId)(Unsafe.unsafe)
+        lastDone       = Ref.unsafe.make[Option[OutDone]](None)(Unsafe.unsafe)
+        errorSignal    = Promise.unsafe.make[Nothing, Unit](fiberId)(Unsafe.unsafe)
+        permits        = Semaphore.unsafe.make(n0)(Unsafe.unsafe)
```

Comment:
What do you think about not using a for-comprehension and use local `val`s for these and avoid all the unnecessary `map`s?

Reply:
I'm not sure to understand your suggestion 🤔

AFAIR, using `v = ...` in a for-comprehension doesn't add any additional `map` call:
```scala
for {
  a <- ZIO.succeed("a")
  b = "b"
  c = "c"
  d <- ZIO.succeed("d")
} yield (a, b, c, d)
```
should produce (something close to if we're using [better-monadic-for](https://github.com/oleg-py/better-monadic-for)):
```scala
ZIO
  .succeed("a")
  .flatMap { a => 
    val b = "b"
    val c = "c"
    ZIO.succeed("d").map(d => (a, b, c, d))
  }
```
no? 🤔

Reply:
Unfortunately, no :( It's Scala version-specific, but in Scala 3 it will produce something along these lines:

```scala
ZIO
  .succeed("a")
  .map { a => 
   
    val b = "b"
    val c = "c"
    (a, b, c)
  }.flatMap { case (a, b, c) =>
    ZIO.succeed("d").map(d => (a, b, c, d))
  }
```

I think in Scala 2 it's very similar or even worse

Reply:
Is your `val` suggestion looking something like this:
```scala
(SingleProducerAsyncInput.make[InErr, InElem, InDone] <*> ZIO.fiberId).flatMap { case (input, fiberId) =>
        val incoming       = ZChannel.fromInput(input)
        val n0             = n.toLong
        val bufferSize0    = bufferSize
        val mergeStrategy0 = mergeStrategy
        val outgoing =
          Queue.unsafe.bounded[ZIO[Env, Either[OutErr, OutDone], OutElem]](bufferSize0, fiberId)(Unsafe.unsafe)
        val cancelers   = Queue.unsafe.unbounded[Promise[Nothing, Unit]](fiberId)(Unsafe.unsafe)
        val lastDone    = Ref.unsafe.make[Option[OutDone]](None)(Unsafe.unsafe)
        val errorSignal = Promise.unsafe.make[Nothing, Unit](fiberId)(Unsafe.unsafe)
        val permits     = Semaphore.unsafe.make(n0)(Unsafe.unsafe)

        for {
          _          <- scope.addFinalizer(outgoing.shutdown)
          _          <- scope.addFinalizer(cancelers.shutdown)
          ...
}
```

? 🤔

Reply:
@kyri-petrou Made an attempt here: https://github.com/zio/zio/pull/9387

LMKWYT